javaparser-core/src/main/java/com/github/javaparser/ast/type/ArrayType.java

```diff
@@ -0,0 +1,26 @@
+package com.github.javaparser.ast.type;
+
+import com.github.javaparser.Range;
+import com.github.javaparser.ast.expr.AnnotationExpr;
+import com.github.javaparser.ast.visitor.GenericVisitor;
+import com.github.javaparser.ast.visitor.VoidVisitor;
+
+import java.util.List;
+
+/**
+ * Every type followed by [] gets wrapped in an ArrayType for each [].
+ */
+public class ArrayType extends ReferenceType {
+    public ArrayType(Range range, Type type, List<AnnotationExpr> annotations) {
```

Comment:
`ArrayType` can't have annotations on its own. The `annotations` field of `ReferenceType` should be moved down to `ClassOrInterfaceType` (which should extend `ReferenceType`).

The JLS states:
> There are four kinds of reference types: class types (§8.1), interface types (§9.1),
type variables (§4.4), and array types (§10.1).